qiskit/transpiler/passes/synthesis/plugin.py

```diff
@@ -166,11 +166,24 @@ def run(self, unitary, **options):
 which defines the interface and contract for high-level synthesis plugins.
 The primary method is
 :meth:`~qiskit.transpiler.passes.synthesis.plugin.HighLevelSynthesisPlugin.run`.
-It takes in a single positional argument, a "higher-level-object" to be
-synthesized, which is any object of type :class:`~qiskit.circuit.Operation`
+The positional argument ``high_level_object`` specifies the "higher-level-object" to
+be synthesized, which is any object of type :class:`~qiskit.circuit.Operation`
 (including, for example,
 :class:`~qiskit.circuit.library.generalized_gates.linear_function.LinearFunction` or
 :class:`~qiskit.quantum_info.operators.symplectic.clifford.Clifford`).
+The positional argument ``target`` specifies the target backend, allowing the plugin
+to access all target-specific information,
+such as the coupling map, the supported gate set, and so on. The positional argument
+``coupling_map`` only specifies the coupling map, and is only used when ``target``
+is not specified.
+The positional argument ``qubits`` specifies the list of qubits over which the
+higher-level-object is defined, in case the synthesis is done on the physical circuit.
```

Comment:
Since you've defined defaults for the arguments they're typically labelled as keyword arguments instead of positional. Although in Python they can be done either way. 

```suggestion
The keyword argument ``target`` specifies the target backend, allowing the plugin
to access all target-specific information,
such as the coupling map, the supported gate set, and so on. The keyword argument
``coupling_map`` only specifies the coupling map, and is only used when ``target``
is not specified.
The keyword argument ``qubits`` specifies the list of qubits over which the
higher-level-object is defined, in case the synthesis is done on the physical circuit.
```